source/as-promise.ts

```diff
@@ -44,21 +51,26 @@ const asPromise = options => {
 			response.body = data;
 
 			try {
-				for (const [index, hook] of Object.entries(options.hooks.afterResponse)) {
-					// eslint-disable-next-line no-await-in-loop
-					response = await hook(response, updatedOptions => {
-						updatedOptions = reNormalize(mergeOptions(options, {
-							...updatedOptions,
-							retry: 0,
-							throwHttpErrors: false
-						}));
-
-						// Remove any further hooks for that request, because we we'll call them anyway.
-						// The loop continues. We don't want duplicates (asPromise recursion).
-						updatedOptions.hooks.afterResponse = options.hooks.afterResponse.slice(0, index);
-
-						return asPromise(updatedOptions);
-					});
+				if (options.hooks && options.hooks.afterResponse) {
```

Comment:
I would prefer using `!` (non-null assertion operator) instead of guarding it here, as we already guarantee its existence, so it's just the TS compiler that is struggling.